examples/exex/kona/src/main.rs

```diff
@@ -0,0 +1,270 @@
+                        attributes,
+                        self.ctx.pool().clone(),
+                        self.ctx.evm_config().clone(),
+                        Arc::clone(&chain_spec),
+                    ).await?;
+                    info!("Executed block with hash: {}", block_hash);
+                    // TODO: fetch block from sequencer and verify block hash matches
+                }
+                Some(notification) = self.ctx.notifications.recv() => {
+                    if let Some(reverted_chain) = notification.reverted_chain() {
+                        reset_sender.send(()).await?;
+                        self.revert(&reverted_chain)?;
+                    }
+
+                    if let Some(committed_chain) = notification.committed_chain() {
+                        self.commit(&committed_chain).await?;
+                        self.ctx.events.send(ExExEvent::FinishedHeight(committed_chain.tip().number))?;
+                    }
+                }
+            }
+            if receiver.is_closed() { break; }
+        }
+
+        Ok(())
+    }
+
+    /// Creates a new derivation pipeline.
+    fn new_pipeline(&self) -> ExExDerivationPipeline {
+        // Place the cursor at the origin.
+        let cursor = L2BlockInfo::default();
+
+        // Create the base rollup config.
+        // TODO: get this from superchain-registry?
+        let rollup_config = Arc::new(RollupConfig::default());
+
+        // Create a stateful attributes provider.
+        let builder = StatefulAttributesBuilder::new(
+            Arc::clone(&rollup_config),
+            ExExL2ChainProvider::new(Arc::clone(&self.l2_chain_provider)),
+            ExExChainProvider::new(Arc::clone(&self.chain_provider)),
+        );
+
+        // Other Providers
+        let reset = ExExResetProvider::new(Arc::clone(&self.tip_state));
+        let l2_chain_provider = ExExL2ChainProvider::new(Arc::clone(&self.l2_chain_provider));
+
+        // Build the ethereum data source
+        let chain_provider = ExExChainProvider::new(Arc::clone(&self.chain_provider));
+        let blob_provider = ExExBlobProvider::new(Arc::clone(&self.blob_provider));
+        let dap_source = EthereumDataSource::new(chain_provider, blob_provider, &rollup_config);
+
+        // Instantiates and link all the stages.
+        let chain_provider = ExExChainProvider::new(Arc::clone(&self.chain_provider));
+        let l1_traversal = L1Traversal::new(chain_provider, Arc::clone(&rollup_config));
+        let l1_retrieval = L1Retrieval::new(l1_traversal, dap_source);
+        let frame_queue = FrameQueue::new(l1_retrieval);
+        let channel_bank = ChannelBank::new(Arc::clone(&rollup_config), frame_queue);
+        let channel_reader = ChannelReader::new(channel_bank, Arc::clone(&rollup_config));
+        let batch_queue = BatchQueue::new(rollup_config.clone(), channel_reader, l2_chain_provider);
+        let queue = AttributesQueue::new(*rollup_config, batch_queue, builder);
```

Comment:
Would be nice if these initializations could be done with a derivation pipeline builder pattern